app/models/filters/filter_base.rb

```diff
@@ -1611,5 +1622,25 @@ def to_comparison
         [start_date, end_date]
       end
     end
+
+    # Strict date parser for filter params.
+    # Accepts:
+    #   - Date objects (returns as-is)
+    #   - nil (returns nil)
+    #   - Strings in 'Mon dd, yyyy' (e.g. 'Apr 27, 2025')
+    #   - Strings in 'YYYY-mm-dd' (e.g. '2025-04-27')
```

Comment:
Can we make it accept and correctly process `mm/dd/yyyy`?  
I think sometimes people type the date in and somehow bypass the JS that should reformat to our format, but given that HMIS is only relevant in the US, it's probably ok to assume they don't mean dd/mm/yyyy.